cli/run_util.go

```diff
@@ -540,14 +583,66 @@ func (r WorkflowRunner) activateAndRunSteps(
 	secrets []envmanModels.EnvironmentItemModel,
 	isLastWorkflow bool,
 	tracker analytics.Tracker,
-	workflowIDProperties coreanalytics.Properties) models.BuildRunResultsModel {
+	workflowIDProperties coreanalytics.Properties,
+	workflowID string,
+) models.BuildRunResultsModel {
 	log.Debug("[BITRISE_CLI] - Activating and running steps")
 
 	if len(workflow.Steps) == 0 {
 		log.Warnf("%s workflow has no steps to run, moving on to the next workflow...", workflow.Title)
 		return buildRunResults
 	}
 
+	err := tools.EnvmanInit(configs.InputEnvstorePath, true)
+	if err != nil {
+		log.Debugf("Couldn't initialize envman.")
+	}
+	err = tools.EnvmanAddEnvs(configs.InputEnvstorePath, *environments)
+	if err != nil {
+		log.Debugf("Couldn't add envs.")
+	}
+	envList, err := tools.EnvmanReadEnvList(configs.InputEnvstorePath)
+	if err != nil {
+		log.Debugf("Couldn't read envs from envman.")
+	}
+
+	serviceContainers, err := r.dockerManager.StartServiceContainers(workflow.Services, workflowID, envList)
+	if err != nil {
+		log.Errorf("❌ Some services failed to start properly!")
+	}
+
+	defer func() {
+		for _, container := range serviceContainers {
+			if err := container.Destroy(); err != nil {
+				log.Errorf("Attempted to stop the docker container for service: %s", container.Name)
+			}
+		}
+	}()
+
+	if workflow.Container.Image != "" {
+		log.Infof("ℹ️ Running workflow in docker container: %s", workflow.Container.Image)
+
+		if err := r.dockerManager.Login(workflow.Container, envList); err != nil {
+			log.Errorf("%s workflow has docker credentials provided, but the authentication failed.", workflow.Title)
+		}
+
+		runningContainer, err := r.dockerManager.StartWorkflowContainer(workflow.Container, workflowID, envList)
+		if err != nil {
+			log.Errorf("Could not start the specified docker image for workflow: %s", workflow.Title)
+		}
+
+		defer func() {
+			if runningContainer == nil {
+				return
+			}
+
+			// TODO: Feature idea, make this configurable, so that we can keep the container for debugging purposes.
```

Comment:
Yes please! Earthly does something similar with an interactive flag for exactly this purpose: https://docs.earthly.dev/docs/guides/debugging

Reply:
At this point we left it as a comment, we can come back to it later if that is okay with you.